packages/googledrive/src/Adaptor.js

```diff
@@ -0,0 +1,257 @@
+
+import { google } from 'googleapis';
+
+let client = undefined;
+
+/**
+ * @private
+ * Creates a connection to Google Drive using OAuth2 authentication.
+ * @param {Object} state - object containing the access token.
+ * @returns {Object} state with Google Drive client initialized.
+ */
+function createConnection(state) {
+  const { accessToken } = state.configuration;
+  const auth = new google.auth.OAuth2();
+  auth.credentials = { access_token: accessToken };
+  client = google.drive({ version: 'v3', auth });
+  return state;
+}
+
+/**
+ * @private
+ * Removes the Google Drive client connection.
+ * @param {Object} state - object.
+ * @returns {Object} state with client reset.
+ */
+function removeConnection(state) {
+  client = undefined;
+  return state;
+}
+
+/**
+ * Execute a sequence of oper.
+ * Wraps `language-common/execute`, and prepends initial state for http.
+ * @example
+ * execute(
+ *   create('foo'),
+ *   delete('bar')
+ * )(state)
+ * @private
+ * @param {Operations} operations - Operations to be performed.
+ * @returns {Operation}
+ */
+export function execute(...operations) {
+  const initialState = {
+    references: [],
+    data: null,
+  };
+
+  return state => {
+    return commonExecute(
+      createConnection,
+      ...operations,
+      removeConnection,
+    )({
+      ...initialState,
+      ...state,
+      configuration: normalizeOauthConfig(state.configuration),
+    });
+  };
+}
+
+/**
+ * Uploads a file to Google Drive with resumable upload support.
+ * @public
+ * @example
+ * create({
+ *   filePath: '/path/to/file.txt',
+ *   fileName: 'custom-name.txt', // optional
+ *   folderId: 'drive-folder-id' // optional
+ * })
+ * @param {Object} params - File upload parameters
+ * @param {string} params.filePath - Local path to the file to upload
+ * @param {string} [params.fileName] - Custom name for the uploaded file
+ * @param {string} [params.folderId] - ID of the parent folder
+ * @returns {Operation} An operation that uploads the file
+ */
+export function create(params) {
+  return state => {
+    const [resolvedParams] = expandReferences(state, params);
+    const { filePath, fileName, folderId } = resolvedParams;
+
+    const fileMetadata = {
+      name: fileName || filePath.split('/').pop(),
+      parents: folderId ? [folderId] : [],
+    };
+
+    const media = {
+      mimeType: 'application/octet-stream',
+      body: fs.createReadStream(filePath),
+    };
+
+    return new Promise((resolve, reject) => {
+      client.files.create(
+        {
+          requestBody: fileMetadata,
+          media: media,
+          fields: 'id,name,webViewLink,size,mimeType,createdTime',
+          uploadType: 'resumable',
+          supportsAllDrives: true, // Important for shared drives
+        },
+        (err, response) => {
+          if (err) {
+            logError(err);
+            if (err.code === 404) {
+              reject(new Error(`Parent folder not found (ID: ${folderId})`));
```

Comment:
Can i suggest you use `util.throwError` instead of `new Error()`  from `language-common`. The errors will be nicely formatted